### PR TITLE
adds file format page

### PR DESCRIPTION
**Pull request by esotrope**
screen shot 2017-08-23 at 3 00 29 pm
screen shot 2017-08-23 at 2 46 46 pm

### CODE REVIEW

File: webapp/src/main/webapp/src/app/groups/import/fileformat/file-format.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { Injectable } from "@angular/core";
+
+@Injectable()
+export class FileFormatService {
+
+  constructor(private http: Http, private translate: TranslateService) {
+  }
+
+  public getTemplateFile(): Observable<any> {
+    return this.http.get('/assets/template/groups-template.csv')
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/groups/import/fileformat/file-format.service.ts:14]: This had to call HTTP directly to avoid the /api prefix.

File: webapp/src/main/webapp/src/app/groups/import/group-import.component.ts [hunk 2]
```diff
@@ -41,15 +41,16 @@ export class GroupImportComponent implements OnInit {
     window.onbeforeunload = this.confirmNavigation.bind(this);
   }
 
-  confirmNavigation(e) {
-    if(this.uploader.isUploading) {
+  confirmNavigation(event) {
+    if (this.uploader.isUploading) {
       let dialogText = this.translate.instant('messages.upload-in-progress');
-      e.returnValue = dialogText;
+      event.returnValue = dialogText;
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/groups/import/group-import.component.ts:47]: unrelated code style/formatting changes

File: webapp/src/main/webapp/src/app/app.routes.ts [hunk 3]
```diff
@@ -19,35 +20,56 @@ export const routes: Routes = [
         path: '',
         pathMatch: 'prefix',
         redirectTo: 'groups'
-      }, {
+      },
+      {
         path: 'groups',
         pathMatch: 'prefix',
         data: { breadcrumb: { translate: 'labels.groups.title' } },
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/app.routes.ts:27]: Interesting thing I found with Will was that the breadcrumb defs need to be on the route entries with paths. Not the ```path: ''``` children. Otherwise the breadcrumb name translation acts goofy.

File: webapp/src/main/webapp/src/app/groups/import/fileformat/file-format.component.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<div>
+  <div class="mb-md">
+    <span class="h3 blue-dark">{{'labels.groups.import.file-format.header' | translate}}</span>
+  </div>
+  <div class="well">
+    <h3>{{'labels.groups.import.file-format.rules.header' | translate}}</h3>
+    <div [innerHtml]="'labels.groups.import.file-format.rules.body-html' | translate"></div>
```
Comment (wtritch) [inline webapp/src/main/webapp/src/app/groups/import/fileformat/file-format.component.html]: innerHTML ?

Comment (esotrope) [reply webapp/src/main/webapp/src/app/groups/import/fileformat/file-format.component.html]: Angular doesn't appear to be picky on this one.

File: webapp/src/main/webapp/src/assets/template/groups-template.csv [hunk 5]
```diff
@@ -0,0 +1 @@
+group_name,school_natural_id,school_year,subject_code,student_ssid,group_user_login
```
Comment (wtritch) [inline webapp/src/main/webapp/src/assets/template/groups-template.csv:1]: Did we want to give them a sample data row or two?  Or just the headers?

**Review by wtritch — APPROVED** [review]

File: webapp/src/main/webapp/src/assets/template/groups-template.csv (hunk 5 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/assets/template/groups-template.csv:1]: Thought about that. Or even giving them a separate "Sample" download beside the "Template" download. As a user I think some hard examples would be useful.

File: webapp/src/main/webapp/src/assets/i18n/en.json [hunk 6]
```diff
@@ -70,6 +70,18 @@
...
+          "header": "CSV File Format",
+          "template": {
+            "button": "Template",
+            "file": "groups.csv"
+          },
+          "rules": {
+            "header": "How to write the CSV",
+            "body-html": "<ol> <li class=\"mt-md\"> The CSV file must include the headers found in the template file provided below. </li><li class=\"mt-md\"> Every row must include a valid <code>school_natural_id</code>, <code>school_year</code>, and <code>group_name</code>. </li><li class=\"mt-md\"> Schools must be grouped together and not separated by rows of groups for other schools. </li><li class=\"mt-md\"> Groups must be grouped together and not separated by rows of other groups. </li><li class=\"mt-md\"> A group's subject must be consistent, and either empty, <code>Math</code>, <code>ELA</code> or <code>All</code>. </li><li class=\"mt-md\"> An empty group subject will be interpreted as <code>All</code>. </li><li class=\"mt-md\"> You cannot assign more than <code>200</code> students to a single group. </li><li class=\"mt-md\"> You must have <code>GROUP_WRITE</code> access to a group's school in order to upload the group. </li></ol>"
+          }
```
Comment (chdietz) [inline webapp/src/main/webapp/src/assets/i18n/en.json:84]: For these; 'grouped together' is that clear, especially 'Groups must be grouped'
  Schools must be grouped together and not separated by rows of groups for other schools. 
  Groups must be grouped together and not separated by rows of other groups.

File: webapp/src/main/webapp/src/assets/template/groups-template.csv (hunk 5 above)
Comment (chdietz) [reply webapp/src/main/webapp/src/assets/template/groups-template.csv:1]: or at least documentation that has a sample in it.

File: webapp/src/main/webapp/src/assets/i18n/en.json (hunk 6 above)
Comment (esotrope) [reply webapp/src/main/webapp/src/assets/i18n/en.json:84]: Hopefully, SBAC will revise the text into what makes sense in academic land.

Comment (chdietz) [reply webapp/src/main/webapp/src/assets/i18n/en.json:84]: also the second line of the warning I would prefer:
Existing groups not in the upload file will remain unchanged.

Comment (esotrope) [reply webapp/src/main/webapp/src/assets/i18n/en.json:84]: Already sent to SBAC for review.
I think I'll hold off changing until we get feedback.
